tests/e2e/automationexercise/api.spec.js

```diff
@@ -0,0 +1,412 @@
+/**
+ * AutomationExercise API Test Suite
+ * Tests the API endpoints provided by AutomationExercise website
+ * @author Guada Gramajo
+ * @date 11-06-25
+ * @requires API endpoints to be accessible
+ */
+
+describe('AutomationExercise - API Tests', () => {
+  const baseUrl = 'https://automationexercise.com'
+  const apiBaseUrl = `${baseUrl}/api`
+
+  beforeEach(() => {
+    // Set up API testing environment
+    cy.intercept('GET', `${apiBaseUrl}/**`).as('getApiCall')
+    cy.intercept('POST', `${apiBaseUrl}/**`).as('postApiCall')
+    cy.intercept('DELETE', `${apiBaseUrl}/**`).as('deleteApiCall')
+  })
+
+  describe('Products API', () => {
+    it('should get all products successfully', () => {
+      cy.request({
+        method: 'GET',
+        url: `${apiBaseUrl}/productsList`,
+        failOnStatusCode: false
+      }).then((response) => {
+        // Verify response status
+        expect(response.status).to.eq(200)
```

Comment:
Great response validation and error handling! Good use of `failOnStatusCode` for robust testing